### PR TITLE
examples/elf : User can select the example which to test.

### PR DESCRIPTION
**Pull request by jeongchanKim**
In wifi app, user can enter the input which test to run.
For recovery, press R or r.

Below is sample output for test.
This is WIFI App

Select Test Scenario.
        -Press R or r : Recovery Test
I'm RECOVERY main! create mkchildren task 9
[9] make_children_task
[10] normal_task
[11] assert_group_main_task
[12] 0th thread, normal thread
[13] 1th thread, normal thread
[14] 2th thread, Write another app space 0x80200100

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5e6969d1fa6a65807da9a3f41c11e3f62bc97d99] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5e6969d1fa6a65807da9a3f41c11e3f62bc97d99] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b8c0e36f681972e9214f17e621496e53dca9a771] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b8c0e36f681972e9214f17e621496e53dca9a771] - Code Rule Check OK.

File: apps/examples/elf/wifiapp/recovery.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	return 0;
+}
+
+static int make_children_task(int argc, char *argv[])
+{
+	int pid;
+
+	printf("[%d] make_children_task \n", getpid());
+
+	pid	= task_create("normal", 100, 1024, normal_task, (FAR char *const *)NULL);
```
Comment (jeongarmy) [inline apps/examples/elf/wifiapp/recovery.c]: please remove tab

File: apps/examples/elf/wifiapp/recovery.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ ****************************************************************************/
+
+void recovery_test(void)
+{
+	int pid;
+
+	pid = task_create("mkchildren", 100, 1024, make_children_task, (FAR char *const *)NULL);
+	if (pid < 0) {
+		printf("task create FAIL\n");
+		return 0;
```
Comment (yashwanth686007) [inline apps/examples/elf/wifiapp/recovery.c]: No need of return. The function return type is "void".

**Comment by seinfra** [comment]
Target : [dd7bc9e7c4877e24213c1b773ec1984b24287ea6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [dd7bc9e7c4877e24213c1b773ec1984b24287ea6] - Code Rule Check (C++) OK.

File: apps/examples/elf/wifiapp/Makefile [hunk 3]
```diff
@@ -92,7 +92,7 @@ BIN_VER = 20190412
 DYNAMIC_RAM_SIZE = 51200
 KERNEL_VER = 2.0
 
-SRCS = wifiapp.c
+SRCS = $(wildcard *.c)
```
Comment (yashwanth686007) [inline apps/examples/elf/wifiapp/Makefile]: I think better to include recovery.c only if EXAMPLES_RECOVERY_TEST is enabled ?

File: apps/examples/elf/wifiapp/wifiapp.c [hunk 4]
```diff
@@ -23,120 +23,43 @@
...
-	while (1);
-
-	return 0;
-}
 
+#include "wifiapp_internal.h"
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+static void display_test_scenario()
```
Comment (sunghan-chang) [inline apps/examples/elf/wifiapp/wifiapp.c]: (void)

File: apps/examples/elf/wifiapp/wifiapp.c [hunk 5]
```diff
@@ -23,120 +23,43 @@
...
+		display_test_scenario();
+		ch = getchar();
+		switch (ch) {
+#ifdef CONFIG_EXAMPLES_RECOVERY_TEST
+		case 'R':
+		case 'r':
+			recovery_test();
+			break;
+#endif
+		default:
```
Comment (sunghan-chang) [inline apps/examples/elf/wifiapp/wifiapp.c]: How about adding 'e' to exit while loop?

**Comment by seinfra** [comment]
Target : [c170732b168e90a387ff34c988503f44f76d2ee6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c170732b168e90a387ff34c988503f44f76d2ee6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c19c54c97b14a96d5d74a9524cc05e1c2251ccd8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c19c54c97b14a96d5d74a9524cc05e1c2251ccd8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0892b55061fc5534a6c4204c61a58a94f8f2b23d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0892b55061fc5534a6c4204c61a58a94f8f2b23d] - Code Rule Check (C++) OK.

**Pull request merged**